src/coreclr/gc/gc.cpp

```diff
@@ -2198,7 +2198,7 @@ CLRCriticalSection gc_heap::check_commit_cs;
 
 size_t      gc_heap::current_total_committed = 0;
 
-size_t      gc_heap::committed_by_oh[total_oh_count] = {0, 0, 0, 0};
+size_t      gc_heap::committed_by_oh[gc_oh_num::total_bucket_count] = {0, 0, 0, 0, 0};
```

Comment:
instead of writing a lot of 0s here, you could use memset to init this with the size like what we do in init_semi_shared for other global variables.